x-pack/plugins/data_visualizer/public/application/common/components/field_type_icon/field_type_icon.tsx

```diff
@@ -6,86 +6,59 @@
  */
 
 import React, { FC } from 'react';
-
 import { EuiToken, EuiToolTip } from '@elastic/eui';
-
 import { i18n } from '@kbn/i18n';
-
 import { getJobTypeAriaLabel } from '../../util/field_types_utils';
-import { JOB_FIELD_TYPES } from '../../../../../common';
 import type { JobFieldType } from '../../../../../common';
 
 interface FieldTypeIconProps {
   tooltipEnabled: boolean;
   type: JobFieldType;
-  fieldName?: string;
   needsAria: boolean;
 }
 
 interface FieldTypeIconContainerProps {
   ariaLabel: string | null;
   iconType: string;
-  color: string;
+  color?: string;
   needsAria: boolean;
   [key: string]: any;
 }
 
+// defaultIcon => a unknown datatype
+const defaultIcon = { iconType: 'questionInCircle', color: 'gray' };
+
+// Extended & modified version of src/plugins/kibana_react/public/field_icon/field_icon.tsx
+export const typeToEuiIconMap: Record<string, { iconType: string; color?: string }> = {
+  boolean: { iconType: 'tokenBoolean' },
+  // icon for an index pattern mapping conflict in discover
+  conflict: { iconType: 'alert', color: 'euiColorVis9' },
+  date: { iconType: 'tokenDate' },
+  date_range: { iconType: 'tokenDate' },
+  geo_point: { iconType: 'tokenGeo' },
+  geo_shape: { iconType: 'tokenGeo' },
+  ip: { iconType: 'tokenIP' },
+  ip_range: { iconType: 'tokenIP' },
+  // is a plugin's data type https://www.elastic.co/guide/en/elasticsearch/plugins/current/mapper-murmur3-usage.html
+  murmur3: { iconType: 'tokenFile' },
+  number: { iconType: 'tokenNumber', color: 'euiColorVis0' },
+  number_range: { iconType: 'tokenNumber', color: 'euiColorVis0' },
+  histogram: { iconType: 'tokenHistogram', color: 'euiColorVis0' },
+  _source: { iconType: 'editorCodeBlock', color: 'gray' },
+  string: { iconType: 'tokenString' },
+  text: { iconType: 'tokenString', color: 'euiColorVis1' },
```

Comment:
We should use `tokenText` here probably, And `tokenKeyword` for the `keyword` type.

Reply:
Updates to the corresponding `EuiToken`s for each field type done in https://github.com/elastic/kibana/pull/112870/commits/858553061dde9609eccf4d50e4ef37ab80391c18 look good.